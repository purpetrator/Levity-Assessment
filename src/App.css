#searchBtn {
  background-color: #6369D1;
  color: white;
  font-size: 20px;
  margin: 50px 20px;

}
#clearBtn {
  background-color: white;
  color: #6369D1;
  min-width: unset;
  outline: none!important;
  border-color: transparent;
  font-size: 25px;
  padding: 20px;
}
.btn {
  border-radius: 6px;
  min-width: 100px;
  padding: 12px 0;
}
.cardsDiv {
  margin: 0 auto;
  width: 50%;
}
.card {
  background-color: #f2f3f8;
  border: none;
  border-radius: 4px;
  min-height: 220px;
  margin: 40px 10px;
}
.card-body {
  color: #5f6266;
  padding: 0.5rem 2rem;
}
.card-title {
  color: #6369D1;
  font-size: 14pt;
  font-weight: bold;
  margin: 20px 0;
  display: flex;
  align-items: center;
}
.owner-avatar {
  height: 20px;
  margin: 3px 5px 0 0;
}
.repo-owner {
  text-decoration: none;
  color: unset;
}
.star-count {
  margin: 10px 0;
  color: #191716;
  font-weight: bold;
}
.star-icon {
  margin: 0 5px 0 0;
  color: #E6AF2E;
}
.card-text {
  font-size: 12pt;
}
.created-at{
  font-size: 10pt;
  margin-top: 30px;
}
.inputField {
  background-color: #f2f3f8;
  border-radius: 6px;
  padding-left: 20px;
  width: 60%;
  display: flex;
}
.mainDiv {
  background-color: white;
  border-radius: 10px;
  margin: 25px auto;
  min-height: 50vh;
  width: 80%;
  padding: 40px 0;
  position: relative;
}
.navbar {
  background-color: #3D348B;
  color: white;
  padding: 15px;
  font-size: 25px;
  display: flex;
  justify-content: center;
  font-weight: bold;
}
.searchBar {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 25px 0;
}
.searchIcon {
  color: #3D348B;
  background-color: #f2f3f8;
  border: none;
  font-size: 16pt;
  margin: 0;
  padding: 0;
  pointer-events: none;
}
body {
  background-color: #f2f3f8;
}
form {
  align-items: center;
  display: flex;
  justify-content: center;
}
input {
  background-color: #f2f3f8;
  border: none;
  border-radius: 6px;
  padding: 15px 20px;
  width: 100%;
  font-size: 20px;
}
input:focus,
button:focus,
button:active,
button:active,
button:hover {
  box-shadow: none !important;
  outline: 0 !important;
}
p {
  font-size: 10pt;
  padding-right: 30px;
}
