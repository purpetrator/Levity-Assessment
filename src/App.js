import {gql, useLazyQuery} from "@apollo/client";
import React, {useState} from "react";
import ResultCard from "./components/ResultCard";
import Nav from "./components/NavBar";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faSearch} from "@fortawesome/free-solid-svg-icons";
import ClearButton from "./components/ClearButton";

import "./App.css";

export const GITHUB_REPOS = gql`
    query findTopRepos($queryString: String!) {
        search(first: 5, query: $queryString, type: REPOSITORY) {
            repositoryCount
            nodes {
                ... on Repository {
                    nameWithOwner
                    createdAt
                    description
                    stargazerCount
                    forkCount
                    name
                    homepageUrl
                    owner {
                        url
                        login
                        avatarUrl(size: 20)
                    }
                    url
                }
            }
        }
    }
`;

export function App() {
    const [query, setQuery] = useState("");

    const [getRepos, {loading, error, data}] = useLazyQuery(GITHUB_REPOS);

    if (error) return `Error! ${error.message}`;

    return (
        <React.Fragment>
            <Nav/>
            <div className="mainDiv">
                <form
                    onSubmit={(event) => {
                        event.preventDefault();
                        getRepos({
                            variables: {queryString: "user:" + query + " sort:stars"},
                        });
                    }}
                    className="searchBar"
                >
                    <div className="inputField">
                        <button className="searchIcon">
                            <FontAwesomeIcon icon={faSearch}/>
                        </button>
                        <input
                            id="myInput"
                            type="text"
                            onChange={(e) => {
                                setQuery(e.target.value);
                            }}
                            placeholder="Search Github repos here"
                        />
                    </div>
                    <ClearButton
                        loading={loading}
                        onClick={() => {
                            document.getElementById("myInput").value = "";
                            getRepos({
                                variables: {queryString: "user:" + "" + " sort:stars"},
                            });
                        }}
                    />
                </form>

                <div className="results">
                    {data?.search?.nodes.map((result, index) => {
                        return (
                            <div
                                className="cardsDiv"
                                id={`${index} ${result.name}`}
                                key={`${index} ${result.name}`}
                            >
                                <ResultCard
                                    name={result.nameWithOwner}
                                    count={result.stargazerCount}
                                    description={result.description}
                                    avatar={result.owner.avatarUrl}
                                    repoURL={result.url}
                                    createdAt={result.createdAt}
                                />
                            </div>
                        );
                    })}
                </div>
            </div>
        </React.Fragment>
    );
}

export default App;
