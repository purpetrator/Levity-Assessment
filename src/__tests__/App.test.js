import React from "react";
import TestRenderer from "react-test-renderer";
import {MockedProvider} from "@apollo/client/testing";
import {App, GITHUB_REPOS} from "../App";

const mockApiCall = [
  {
    request: {
      query: GITHUB_REPOS,
      variables: {
        queryString: "user:shopify sort:stars",
      },
    },
    result: {
      data: {
        search: {
          repositoryCount: 369,
          nodes: [
            {
              nameWithOwner: "Shopify/draggable",
              createdAt: "2017-09-27T15:31:23Z",
              description:
                "The JavaScript Drag & Drop library your grandparents warned you about.",
              stargazerCount: 15811,
              forkCount: 1013,
              name: "draggable",
              homepageUrl: "https://shopify.github.io/draggable",
              owner: {
                url: "https://github.com/Shopify",
                login: "Shopify",
                avatarUrl:
                  "https://avatars.githubusercontent.com/u/8085?s=20&v=4",
                __typename: "Organization",
              },
              url: "https://github.com/Shopify/draggable",
              __typename: "Repository",
            },
            {
              nameWithOwner: "Shopify/dashing",
              createdAt: "2012-07-18T21:01:40Z",
              description:
                "The exceptionally handsome dashboard framework in Ruby and Coffeescript.",
              stargazerCount: 11076,
              forkCount: 1230,
              name: "dashing",
              homepageUrl: "http://shopify.github.com/dashing/",
              owner: {
                url: "https://github.com/Shopify",
                login: "Shopify",
                avatarUrl:
                  "https://avatars.githubusercontent.com/u/8085?s=20&v=4",
                __typename: "Organization",
              },
              url: "https://github.com/Shopify/dashing",
              __typename: "Repository",
            },
            {
              nameWithOwner: "Shopify/liquid",
              createdAt: "2008-05-08T15:27:26Z",
              description:
                "Liquid markup language. Safe, customer facing template language for flexible web apps. ",
              stargazerCount: 9330,
              forkCount: 1226,
              name: "liquid",
              homepageUrl: "https://shopify.github.io/liquid/",
              owner: {
                url: "https://github.com/Shopify",
                login: "Shopify",
                avatarUrl:
                  "https://avatars.githubusercontent.com/u/8085?s=20&v=4",
                __typename: "Organization",
              },
              url: "https://github.com/Shopify/liquid",
              __typename: "Repository",
            },
            {
              nameWithOwner: "Shopify/sarama",
              createdAt: "2013-07-05T18:52:38Z",
              description: "Sarama is a Go library for Apache Kafka.",
              stargazerCount: 8180,
              forkCount: 1413,
              name: "sarama",
              homepageUrl: "https://shopify.github.io/sarama",
              owner: {
                url: "https://github.com/Shopify",
                login: "Shopify",
                avatarUrl:
                  "https://avatars.githubusercontent.com/u/8085?s=20&v=4",
                __typename: "Organization",
              },
              url: "https://github.com/Shopify/sarama",
              __typename: "Repository",
            },
            {
              nameWithOwner: "Shopify/toxiproxy",
              createdAt: "2014-09-04T13:56:38Z",
              description:
                ":alarm_clock: :fire: A TCP proxy to simulate network and system conditions for chaos and resiliency testing",
              stargazerCount: 7806,
              forkCount: 360,
              name: "toxiproxy",
              homepageUrl: "http://toxiproxy.io",
              owner: {
                url: "https://github.com/Shopify",
                login: "Shopify",
                avatarUrl:
                  "https://avatars.githubusercontent.com/u/8085?s=20&v=4",
                __typename: "Organization",
              },
              url: "https://github.com/Shopify/toxiproxy",
              __typename: "Repository",
            },
          ],
          __typename: "SearchResultItemConnection",
        },
      },
    },
  },
];

it("successfully renders some HTML", () => {
  const component = TestRenderer.create(
    <MockedProvider mocks={mockApiCall} addTypename={false}>
      <App />
    </MockedProvider>
  );

  expect(component).toMatchSnapshot();
});
