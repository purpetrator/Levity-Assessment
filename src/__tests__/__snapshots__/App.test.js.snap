// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully renders some HTML 1`] = `
<div>
  <nav
    className="navbar"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand mb-0 h1"
      >
        Github Repository Search
      </span>
    </div>
  </nav>
  <div
    className="mainDiv"
  >
    <form
      className="searchBar"
      onSubmit={[Function]}
    >
      <div
        className="inputField"
      >
        <button
          className="searchIcon"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
        <input
          id="myInput"
          onChange={[Function]}
          placeholder="Search Github repos here"
          type="text"
        />
      </div>
      <button
        className="btn"
        id="clearBtn"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-ban "
          data-icon="ban"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.034 8 8 119.033 8 256s111.034 248 248 248 248-111.034 248-248S392.967 8 256 8zm130.108 117.892c65.448 65.448 70 165.481 20.677 235.637L150.47 105.216c70.204-49.356 170.226-44.735 235.638 20.676zM125.892 386.108c-65.448-65.448-70-165.481-20.677-235.637L361.53 406.784c-70.203 49.356-170.226 44.736-235.638-20.676z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </form>
    <div
      className="results"
    />
  </div>
</div>
`;
